interfaces/webservices.go

```diff
@@ -0,0 +1,302 @@
+package interfaces
+
+import (
+	"fmt"
+	"io"
+	"net/http"
+	"strconv"
+
+	"game-tracker/domain"
+	"game-tracker/usecases"
+)
+
+type ProfileInteractor interface {
+	ShowLibrary(userId, libraryId int) (string, []domain.Game, error)
+	AddUser(player domain.Player, userName string) error
+	RemoveUser(playerId, userId int) error
+	EditUserInfo(userId int, info string) error
+	AddGame(userId, libraryId int, gameName, gameProducer string, gameValue float64) error
+	RemoveGame(userId, libraryId, gameId int) error
+}
+
+type WebserviceHandler struct {
+	ProfileInteractor usecases.ProfileInteractor
+}
+
+func (handler WebserviceHandler) ShowLibrary(res http.ResponseWriter, req *http.Request) error {
+	userId, err := getFormUserId(req)
+	if err != nil {
+		return err
+	}
+	libraryId, err := getFormLibraryId(req)
+	if err != nil {
+		return err
+	}
+
+	info, games, err := handler.ProfileInteractor.ShowLibrary(userId, libraryId)
+	if err != nil {
+		return err
+	}
+
+	io.WriteString(res, fmt.Sprintf("Library #%d of user #%d\n", libraryId, userId))
```

Comment:
Well, hard to explain. You can look at its code
https://golang.org/src/io/io.go?s=9764:9819#L267
It calls function `Write` of type `Writer interface` or `WriteString` of `stringWriter interface`
https://golang.org/pkg/io/#Writer

> It returns the number of bytes written from p (0 <= n <= len(p)) and any error encountered that caused the write to stop early

Because of interface, so we will not know what will be implemented. But if a function return `error`, we should not ignore it.